### PR TITLE
fix: data sections in test wat codes

### PR DESCRIPTION
**Pull request by jakmeier**
Contracts with a controlled data section size are generated ad-hoc for
tests and estimations. But they never actually worked, they hit
`PrepareError::Deserialization` instead.

Test plan
-----------------
Validate the generated wat codes using `prepare_contract`.

### CODE REVIEW

**Comment by jakmeier** [comment]
I realised that these modules never actually compiled. My tests so far still (mostly) worked as intended, since I only needed the code to be of a certain size.

But I'm a bit worried that I didn't even notice the compilation error, as deploying invalid code is not a visible failure mode. I only noticed it now because I was trying to invoke `main` on the deployed contract.

So I guess perhaps @nagisa, is there an easy way to add an assertion, or something along those lines, that would make sure the output of `wat::parse_str(wat_code).unwrap()` is actually valid?

**Comment by nagisa** [comment]
Wouldn't running the wasm code through preparation (cf. `prepare_contract`) do that?

**Comment by matklad** [comment]
>But I guess to properly validate the wasm, we need to go through near-vm-runner code.

We can add a test for this to `near-vm-runner`. Seems a bit more targeted than adding checks to a bunch of integrated tests. No super strong opinion here, fine either way!

File: integration-tests/src/tests/client/process_blocks.rs [hunk 1]
```diff
@@ -3501,6 +3502,8 @@ fn test_deploy_cost_increased() {
 
     let contract_size = 1024 * 1024;
     let test_contract = near_test_contracts::sized_contract(contract_size);
+    // Rune code through preparation for validation. (Deploying will succeed either way).
```
Comment (nagisa) [inline integration-tests/src/tests/client/process_blocks.rs:3505]: Typo?

```suggestion
    // Run code through preparation for validation. (Deploying will succeed either way).
```

**Comment by jakmeier** [comment]
> > But I guess to properly validate the wasm, we need to go through near-vm-runner code.
> 
> We can add a test for this to `near-vm-runner`. Seems a bit more targeted than adding checks to a bunch of integrated tests. No super strong opinion here, fine either way!

That would have been a cleaner solution indeed. Thanks for the input!

Since it is already done like this, I will just merge it as-is. But I will keep that approach in mind next time I run into this problem. :)

**Pull request merged**